src-gui/src/main/java/org/openkilda/service/StatsService.java

```diff
@@ -252,51 +253,74 @@ public String getFlowLossPacketStats(String startDate, String endDate, String do
     }
 
     /**
-     * Gets the flow path stat.
+     * Retrieves Victoria Flow Path statistics data based on the provided request parameters.
      *
-     * @param flowPathStats the flow path stat
-     * @return the flow path stat
+     * @param statsReq The request parameters for querying Victoria Flow Path statistics.
+     * @return A list of VictoriaData objects representing the queried statistics.
+     * @throws InvalidRequestException if the request parameters are invalid or if there are issues with the query.
      */
-    public String getFlowPathStats(FlowPathStats flowPathStats) {
-        return statsIntegrationService.getStats(flowPathStats.getStartDate(), flowPathStats.getEndDate(),
-                flowPathStats.getDownsample(), getSwitches(flowPathStats), null, flowPathStats.getFlowid(),
-                null, flowPathStats.getInPort(), null, flowPathStats.getOutPort(),
-                StatsType.FLOW_RAW_PACKET, flowPathStats.getMetric(), flowPathStats.getDirection());
+    public List<VictoriaData> getVictoriaStats(VictoriaStatsReq statsReq) throws InvalidRequestException {
+        validateRequestParameters(statsReq.getStartDate(), statsReq.getStatsType(), statsReq.getMetrics());
+
+        Long startTimeStamp = parseTimeStamp(statsReq.getStartDate());
+        Long endTimeStamp = Optional.ofNullable(parseTimeStamp(statsReq.getEndDate()))
+                .orElse(System.currentTimeMillis() / 1000);
+
+        List<VictoriaData> victoriaDataList = new ArrayList<>();
+        List<String> metricNameList = Lists.newArrayList();
```

Comment:
why the two different array list declarations?

Reply:
fixed